pkg/domain/infra/abi/containers.go

```diff
@@ -264,25 +264,27 @@ func (ic *ContainerEngine) ContainerRm(ctx context.Context, namesOrIds []string,
 	reports := []*entities.RmReport{}
 
 	names := namesOrIds
-	for _, cidFile := range options.CIDFiles {
-		content, err := ioutil.ReadFile(cidFile)
-		if err != nil {
-			return nil, errors.Wrap(err, "error reading CIDFile")
-		}
-		id := strings.Split(string(content), "\n")[0]
-		names = append(names, id)
-	}
-
 	// Attempt to remove named containers directly from storage, if container is defined in libpod
 	// this will fail and code will fall through to removing the container from libpod.`
 	tmpNames := []string{}
 	for _, ctr := range names {
 		report := entities.RmReport{Id: ctr}
-		if err := ic.Libpod.RemoveStorageContainer(ctr, options.Force); err != nil {
+		report.Err = ic.Libpod.RemoveStorageContainer(ctr, options.Force)
+		switch errors.Cause(report.Err) {
+		case nil:
 			// remove container names that we successfully deleted
-			tmpNames = append(tmpNames, ctr)
-		} else {
 			reports = append(reports, &report)
+		case define.ErrNoSuchCtr:
+			// remove container names that does not exist
+			reports = append(reports, &report)
+		default:
+			if _, err := ic.Libpod.LookupContainer(ctr); errors.Cause(err) == define.ErrNoSuchCtr {
```

Comment:
This bit seems unnecessary - I don't see any way we fail in a fashion that's not ErrNoSuchCtr before `RemoveStorageContainer` checks Libpod to see if the container exists.

Reply:
The use case we are trying to fix #8735, is the case where we created a buildah container that is in use, and fails.  This would end up in this block, and now we check to see if it is a libpod container.  Since it is not, we return the error. 
Thus we end up with an error telling us the container is in use, and the user has to add --force.
In the current code we fail over to trying to remove a libpod container, and we report
container does not exist.

Reply:
I'm not sure you're right here - the check for whether a container is a Libpod container in `RemoveStorageContainer` happens before we try to remove the container, so we should already have gotten an `ErrCtrExists`

Reply:
If you run @edsantiago test you will see this happen.

Reply:
Then we need to catch this inside `RemoveStorageContainer` - why work around broken behavior when we can just fix the broken API

Reply:
RemoveStorageContainer is not broken, it is returning a valid error, that we are returning to the user.  The problem is we are ignoring the error and then falling through to the libpod remove, which finds that there is no Container libpod container, and reports container does not exists.